cudaaligner/src/alignment.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+* Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+*
+* NVIDIA CORPORATION and its licensors retain all intellectual property
+* and proprietary rights in and to this software, related documentation
+* and any modifications thereto.  Any use, reproduction, disclosure or
+* distribution of this software and related documentation without an express
+* license agreement from NVIDIA CORPORATION is strictly prohibited.
+*/
+
+#include <claragenomics/cudaaligner/alignment.hpp>
+
+#include <iostream>
+
+namespace claragenomics
+{
+
+namespace cudaaligner
+{
+
+std::ostream& operator<<(std::ostream& os, const FormattedAlignment& formatted_alignment)
+{
+    std::size_t line_length              = (formatted_alignment.linebreak_after == 0) ? formatted_alignment.query.size() : formatted_alignment.linebreak_after;
+    std::size_t number_of_complete_lines = (std::size_t)(formatted_alignment.query.size() / line_length);
+    for (std::size_t i = 0; i < number_of_complete_lines; i++)
```

Comment:
Good logic, but you should be able to simplify this function:
I looked up `std::string::substr(pos, length)`. It's fine if `length` is beyond the end of the string. It will just return the remainder then.
So how about just one loop
```
for (std::size_t i=0; i < formatted_alignment.query.size(); i+=line_length)
{
...substr(i, line_length)...
}
```
?